RHIDP-3440-install CX: Add plugin install content (#547)
redhat-developer/red-hat-developers-documentation-rhdh: modules/dynamic-plugins/con-dynamic-plugins-cache.adoc

[id="con-dynamic-plugin-cache_{context}"]

> = Using the dynamic plugins cache

Author: Moving this to the Configure title.

> The dynamic plugins cache feature in {product} ({product-very-short}) optimizes the installation process of dynamic plugins and reduces platform boot time. This feature caches previously installed plugins, eliminating the need to re-download them on subsequent boots if their configuration has not changed.

Reviewer: ```suggestion
The dynamic plugins cache in {product} ({product-very-short}) enhances the installation process and reduces platform boot time by storing previously installed plugins. If the configuration remains unchanged, this feature prevents the need to re-download plugins on subsequent boots.
```

> When you enable dynamic plugins cache, the feature works as follows:

Reviewer: ```suggestion
When you enable dynamic plugins cache:
```

* The system calculates a checksum of each plugin's YAML configuration (excluding `pluginConfig`).

> * This checksum is stored in a file named `dynamic-plugin-config.hash` within the plugin's directory.

Reviewer: ```suggestion
* The checksum is stored in a file named `dynamic-plugin-config.hash` within the plugin's directory.
```

* During boot, if a plugin's package reference matches the previous installation and the checksum is unchanged, the download is skipped.

> * Plugins that have been disabled since the previous boot are automatically removed.

Reviewer: ```suggestion
* Plugins that are disabled since the previous boot are automatically removed.
```

> == Enabling the dynamic plugins cache

Reviewer: I suggest applying the mod-docs procedural module template. It would be helpful to include prerequisites and a procedure tag with an ordered list of steps.

To enable the dynamic plugins cache in RHDH, the plugins directory `dynamic-plugins-root` must be a persistent volume. 

For Helm chart installations, a persistent volume named `dynamic-plugins-root` is automatically created.

For operator-based installations, you must manually create the PersistentVolumeClaim (PVC) as follows:

[source,yaml]
----
kind: PersistentVolumeClaim
apiVersion: v1
metadata:
  name: dynamic-plugins-root
spec:
  accessModes:
    - ReadWriteOnce
  resources:
    requests:
      storage: 5Gi

---

apiVersion: rhdh.redhat.com/v1alpha2
kind: Backstage
metadata:
  name: rhdh
spec:
  deployment:
    patch:
      spec:
        template:
          spec:
            volumes:
              - $patch: replace
                name: dynamic-plugins-root
                persistentVolumeClaim:
                  claimName: dynamic-plugins-root
----

[NOTE]
Future versions of the RHDH operator are planned to automatically create the PVC.

> == Configuring the dynamic plugins cache

Reviewer: Similarly, can we apply the procedural module template in this section too?
